### ISSUE TITLE
Support Electron

### ISSUE THREAD

**Issue by duanyao** [whiteout-io/tcp-socket#23]
[Electron](https://github.com/atom/electron/), which is similar to NW.js, has node.js `require()` but not `define()`, so currently is not supported.

If change 
```JS
    else if (typeof define === 'function' && define.amd && typeof nodeRequire !== 'undefined') {
        // amd under node-webkit
        define([], factory.bind(null, navigator, null, nodeRequire('net'), nodeRequire('tls')));
    }
```
to
```JS
    else if (typeof nodeRequire !== 'undefined') {
        factory(navigator, null, nodeRequire('net'), nodeRequire('tls'));
    }
```
it works in Electron.

It seems `define()` is not necessary even in NW.js?

**Comment by felixhammerl** [comment]
the whole header was pretty stupid, i will put up a fix right away

**Comment by felixhammerl** [comment]
Fixed with https://github.com/whiteout-io/tcp-socket/commit/dac0697cf04aa03c09ddd25612dd74641e11282f

**Comment by felixhammerl** [comment]
published as v0.5.6
